app/controllers/provider_interface/decisions_controller.rb

```diff
@@ -14,7 +14,36 @@ def submit_response
     end
 
     def new_offer
-      raise 'Not yet implemented'
+      @application_offer = MakeAnOffer.new(application_choice: @application_choice)
+    end
+
+    def confirm_offer
+      standard_conditions_array = params.dig(:make_an_offer, :standard_conditions)
+      further_conditions_array = [params.dig(:make_an_offer, :first_condition), params.dig(:make_an_offer, :second_condition), params.dig(:make_an_offer, :third_condition), params.dig(:make_an_offer, :further_condition)].reject(&:blank?)
+      complete_conditions_array = [standard_conditions_array, further_conditions_array].compact.reduce([], :|)
+      @application_offer = MakeAnOffer.new(
+        application_choice: @application_choice,
+        offer_conditions: complete_conditions_array,
+      )
+      render action: :new_offer if !@application_offer.valid?
+    end
+
+    def create_offer
+      offer_conditions_array = JSON.parse(params.dig(:offer_conditions))
+
+      @application_offer = MakeAnOffer.new(
+        application_choice: @application_choice,
+        offer_conditions: offer_conditions_array,
+      )
+
+      if @application_offer.save
+        flash[:success] = 'Application status changed to ‘Offer’'
```

Comment:
Not keen on this success message. What's the effect of changing the status? Do we need to communicate more?

Reply:
Changed to use en locale for displaying status. Should sort this out 👍 
updated here: https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/555/commits/04095d4763930cf8266f569bf6a7e96b5b17d370